rfc/openfed__prerequisite.md

```diff
@@ -0,0 +1,408 @@
+(`ReviewInput.productUpc`).
+```graphql
+type Mutation {
+  addReviewToProduct(
+    input: ReviewInput!
+    @openfed__prerequisite(resolveEntity: { typeName: "Product", fields: "sku upc:productUpc" })
+  ): Review
+}
+
+input ReviewInput {
+  content: String!
+  sku: ID!
+  productUpc: Int!
+}
+
+type Review {
+  content: String!
+}
+
+type Product @key(fields: "sku upc") {
+  sku: ID!
+  upc: Int!
+  reviews: [Review!]!
+}
+```
+
+4. Valid: Key field name `upc` directly corresponds to a top-level Input field (`ReviewInput.upc`). 
+Key field name `usku` does not directly correspond to a top-level Input field but defines valid mapping 
+(`ReviewInput.nestedInput.sku`).
+```graphql
+type Mutation {
+  addReviewToProduct(
+    input: ReviewInput!
+    @openfed__prerequisite(resolveEntity: { typeName: "Product", fields: "sku:nestedInput.sku upc" })
+  ): Review
+}
+
+input ReviewInput {
+  content: String!
+  upc: Int!
+  nestedInput: NestedInput!
+}
+
+input NestedInput {
+  sku: ID!
+}
+
+type Review {
+  content: String!
+}
+
+type Product @key(fields: "sku upc") {
+  sku: ID!
+  upc: Int!
+  reviews: [Review!]!
+}
+```
+
+5. Invalid (composition error): Key field name `sku` defines invalid mapping (`ReviewInput.nestedInput.sku` does not 
+exist).
+```graphql
+type Mutation {
+  addReviewToProduct(
+    input: ReviewInput!
+    @openfed__prerequisite(resolveEntity: { typeName: "Product", fields: "sku:nestedInput.sku upc" })
+  ): Review
+}
+
+input ReviewInput {
+  content: String!
+  upc: Int!
+  nestedInput: NestedInput!
+}
+
+input NestedInput {
+  productSku: ID!
+}
+
+type Review {
+  content: String!
+}
+
+type Product @key(fields: "sku upc") {
+  sku: ID!
+  upc: Int!
+  reviews: [Review!]!
+}
+```
+
+6. Invalid (composition error): All key field names define valid mapping but `NestedInput.productSku` of type `[ID]`
+is incompatible with type `ID!`.
+```graphql
+type Mutation {
+  addReviewToProduct(
+    input: ReviewInput!
+    @openfed__prerequisite(resolveEntity: { typeName: "Product", fields: "sku:nestedInput.productSku upc:productUpc" })
+  ): Review
+}
+
+input ReviewInput {
+  content: String!
+  productUpc: Int!
+  nestedInput: NestedInput!
+}
+
+input NestedInput {
+  productSku: Int!
+}
+
+type Review {
+  content: String!
+}
+
+type Product @key(fields: "sku upc") {
+  sku: ID!
+  upc: Int!
+  reviews: [Review!]!
+}
+```
+
+### Entity Representation Call returns data
+If the entity representation call is successful, data will be returned.
+This means that entity reference is valid and the operation cannot proceed.
+
+### Entity Representation Call returns null
+If the entity representation call is unsuccessful, null will be returned.
+This means that entity reference is invalid and operation cannot proceed.
+Cosmo will produce an error and the operation will not be committed.
+
+### Potential discussion points
+- Multiple entities (batching)
```

Comment:
This seems very important. Is this API intended to only work for mutations or is this usable for queries as well?